### PR TITLE
feat: add url /robots.txt

### PR DESCRIPTION
**Pull request by snxq**
**What is the problem I am trying to address?**

add `/robots.txt` to allow or disallow web crawler. 

**Mention the issue number it fixes or add the details of the changes if it doesn't have a specific issue.**

Fixes #

#1264

### CODE REVIEW

File: cmd/proxy/actions/robots.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import (
+	"io/ioutil"
+	"net/http"
+
+	"github.com/gomods/athens/pkg/errors"
+)
+
+func robotsHandler(f string) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		b, err := ioutil.ReadFile(f)
```
Comment (marwan-at-work) [inline cmd/proxy/actions/robots.go]: use `http.ServeFile` instead as it will write proper headers.

File: pkg/config/config.go [hunk 2]
```diff
@@ -44,6 +44,7 @@ type Config struct {
 	SumDBs           []string `envconfig:"ATHENS_SUM_DBS"`
 	NoSumPatterns    []string `envconfig:"ATHENS_GONOSUM_PATTERNS"`
 	SingleFlightType string   `envconfig:"ATHENS_SINGLE_FLIGHT_TYPE"`
+	RobotsFile       string   `envconfig:"ROBOTS_FILE"`
```
Comment (marwan-at-work) [inline pkg/config/config.go]: this needs to be propagated in a few places: `config.dev.toml`, `helm` and `defaultConfigs`

**Comment by arschles** [comment]
@snxq following up on what @marwan-at-work said, are you seeing a need to have Athens serve a customized robots.txt file? If not, do you think his suggestion of serving a static page would work?

**Comment by snxq** [comment]
@marwan-at-work @arschles 
Sorry for replying so late. understood your point.I will change this to serve a static page.

**Comment by arschles** [comment]
@snxq did you mean to close this without getting it reviewed and merged? I think @marwan-at-work was saying that we can serve a static response for `/robots.txt`, and we don't need that to be configurable (@marwan-at-work correct me if I'm wrong)

**Comment by snxq** [comment]
@arschles Sorry about this. I was a little busy before.So I close this pr.
I can reopen this and resubmit if we still need it.

**Comment by arschles** [comment]
@snxq no problem. if you end up seeing a need for a `robots.txt` file, feel free to submit another PR!

**Comment by snxq** [comment]
@arschles okey.

**Pull request closed without merging**